File: src/zoo/Cat.java
package zoo;
public class Cat extends Feline {
	
	public Cat(String name) {
		super(name);
	}
	
	
	/* creates random number from [0,1] , and if that number is below or equal to 0.5, 
	 * invoke makeNoise1() otherwise, invoke makeNoise2()
	 */
	
	public void makeNoise() {
		double x = Math.random();
		
//		System.out.println(x);
		
		if(x <= 0.5) {
			makeNoise1();
		} else {
			makeNoise2();
		}
	}
	
	public void makeNoise1() {
		System.out.println("My type is " + this.getClass() +  " " + this.name + " Me-yao");
	}
	
	public void makeNoise2() {
		System.out.println("My type is " + this.getClass() + " " + this.name + " Hello");
	}
	
	// Reference: https://dzone.com/articles/random-number-generation-in-java
	// for random number generator1
	
}

File: python-code/Cat.py
import random
from Feline import Feline 
from Animals import *

class Cat(Feline):
    def __init__(self,name):
        self.name = name
        self.sound = hissNoise()
        

    # def makeNoise(self):
    #     x = random.random()
    #     if(x <= 0.5):
    #         self.makeNoise1()
    #     else: 
    #         self.makeNoise2()

    def makeNoise1(self):
        print("My type is", self.__class__.__name__, " ", self.name, "Me_yao")

    def makeNoise2(self):
        print("My type is", self.__class__.__name__, " ", self.name, "Hello")
